lambda/lambda.mli

```diff
@@ -195,6 +195,10 @@ type primitive =
                         one; O(1) *)
   | Parray_of_iarray (* Unsafely reinterpret an immutable array as a mutable
                         one; O(1) *)
+  | Pempty_iarray (* The equivalent of [[::]], but available without any
```

Comment:
1. I use it because we need to create empty arrays in `stdlib/iarray.ml`.
2. I don't know how to get the build system to do that; I'd be perfectly happy to.
3. Before this, I cast `[||]` to `[::]` in `stdlib/iarray.ml`; now I have to use it directly, because I have to be able to put it on the local stack.
4. We can't use the expanded forms because the parser can't tell the difference and is still unhappy.

Reply:
These are all good answers. But I'd still say all of the fuss in this patch to make a new primitive isn't worth it. I see the following possible ways forward:

- The manual documents `ocaml_compiler_internal_params` (see just before https://v2.ocaml.org/manual/comp.html#s:modules-file-system). That looks like it does what you want to enable an extension for an individual file. It's processed in `driver/compenv.ml` in the `apply_config_file` function. It looks like you'd have to reverse-engineer the file format a bit, but it doesn't look too hard.
- I think it's actually OK to have an extension on for the entire stdlib, if the above suggestion doesn't pan out. I'm not exactly sure where to put the extension, but `stdlib/Makefile` has `COMPFLAGS` which looks like a very good starting point. Not sure where the dune equivalent is, or if it's necessary.
- I seem to recall some discussion of a way of enabling an extension in a file. But I can't find evidence of this. Perhaps ask around on Slack.
- Failing all that, I think adding a new C function that just creates an empty array is still much less invasive than a whole new primitive.
- And, actually, I don't see a great reason not to just make an empty normal array and cast it to an immutable one. I seem to recall that empty arrays are shared on the heap, so it wouldn't do any allocation anyway. (Double-check this.)

Reply:
Do I understand correctly that you are trying to make a local empty immutable array? I think you can continue to just cast a `[||]` because that is never an actual allocation at runtime.

Reply:
Not only was this sensible advice, but it turns out we didn't actually use any local `[::]`s anymore :man_facepalming:.  `%empty_iarray` is gone again.